src/applications/soundDetectionDemo/robotMovement/src/vRobotMovement.cpp

```diff
@@ -0,0 +1,369 @@
+
+    // Joint steep
+    double joint_step;
+    double half_joint_step;
+
+    // Joint initial position
+    double joint_initial_position;
+
+    // Joint current position
+    double joint_current_position;
+
+    // Joint desired position
+    double joint_desired_position;
+
+    // Period
+    double update_period;
+
+public:
+
+    vRobotMovement() {}
+
+    virtual bool configure(yarp::os::ResourceFinder& rf)
+    {
+        yInfo() << "Configuring the module...";
+
+        // Set the module name used to name ports
+        setName((rf.check("name", Value("/vRobotMovement")).asString()).c_str());
+
+        // Open input port
+        yInfo() << "Opening input port...";
+        if(!input_port.open(getName() + "/AE:i")) {
+            yError() << "Could not open input port";
+            return false;
+        }
+        // Open output port
+        yInfo() << "Opening output port...";
+        if(!output_port.open(getName() + "/pos:o")) {
+            yError() << "Could not open output port";
+            return false;
+        }
+
+        // Read flags and parameters
+        
+        // Debug flag: flase by default
+        is_debug_flag = rf.check("is_debug_flag") &&
+                rf.check("is_debug_flag", Value(true)).asBool();
+        yInfo() << "Flag is_debug_flat is: " << is_debug_flag;
+
+        // Simulation flag: true by default
+        bool default_is_simulationr = true;
+        is_simulation = rf.check("is_simulation") &&
+                rf.check("is_simulation", Value(default_is_simulationr)).asBool();
+        yInfo() << "Flag is_simulation is: " << is_simulation;
+
+        // Number of sound source positions to classify: 9 by default
+        int default_number_sound_source_neurons = 16;
+        number_sound_source_neurons = rf.check("number_sound_source_neurons",
+                                    Value(default_number_sound_source_neurons)).asInt();
+        yInfo() << "Setting number_sound_source_neurons parameter to: " << number_sound_source_neurons;
+
+        // iCub body part to be moved: "head" by default
+        std::string default_body_part = "head";
+        body_part = rf.check("body_part",
+                                    Value(default_body_part)).asString();
+        yInfo() << "Setting body_part parameter to: " << body_part;
+
+        // Joint ID to indicate the joint to be controlled: 2 by default
+        int default_joint_id = 2;
+        joint_id = rf.check("joint_id",
+                                    Value(default_joint_id)).asInt();
+        yInfo() << "Setting joint_id parameter to: " << joint_id;
+
+        // Joint speed of movement: 10 by default
+        double default_joint_speed = 10.0;
+        joint_speed = rf.check("joint_speed",
+                                    Value(default_joint_speed)).asDouble();
+        yInfo() << "Setting joint_speed parameter to: " << joint_speed;
+
+        // Period value: 1 sec by default
+        double default_update_period = 1.0;
```

Comment:
is there a reason you are defining a variable for the default and then assigning it in the rf.check() and not just writing the default value in the rf.check() itself?